packages/eslint-plugin-react-hooks/src/ExhaustiveDeps.js

```diff
@@ -197,6 +182,47 @@ export default {
         }
       }
 
+      // Warn about accessing .current in cleanup effects.
+      currentRefsInEffectCleanup.forEach(
+        ({reference, dependencyNode}, dependency) => {
+          const references = reference.resolved.references;
+          // Is React managing this ref or us?
+          // Let's see if we can find a .current assignment.
+          let foundCurrentAssignment = false;
+          for (let i = 0; i < references.length; i++) {
+            const {identifier} = references[i];
+            const {parent} = identifier;
+            if (
+              parent != null &&
+              // ref.current
+              parent.type === 'MemberExpression' &&
+              !parent.computed &&
+              parent.property.type === 'Identifier' &&
+              parent.property.name === 'current' &&
+              // ref.current = <something>
+              parent.parent.type === 'AssignmentExpression' &&
+              parent.parent.left === parent
+            ) {
+              foundCurrentAssignment = true;
+              break;
+            }
+          }
+          // We only want to warn about React-managed refs.
+          if (foundCurrentAssignment) {
+            return;
+          }
+          context.report({
+            node: dependencyNode.parent.property,
+            message:
+              `Accessing '${dependency}.current' during ` +
+              `the effect cleanup is likely a mistake because by this time React ` +
```

Comment:
A cultural suggestion: Explain what is happening and why rather than telling someone directly they made a mistake. I suspect people might get overly defensive if the computer tells them they made a mistake.
```suggestion
              `the effect cleanup will likely cause a TypeError because by this time React ` +
```
No opinion what should we call it e.g. "null pointer exception", "reading a property from `null`" etc.

Reply:
TypeError isn't the main problem (easy to work around). A bigger problem is that you'll always keep getting the *previous* ref value since it's being read too early. The challenge is to explain this without freaking people out about how "complex" it is.

Reply:
Oops, I misread. This is a different warning from the one I was thinking about. In this case the problem is that you're reading the *next* ref value when you meant to access the *previous* one.

Reply:
Right, maybe 
```suggestion
              `the effect cleanup will likely use a different ref value because by this time React ` +
```

My point was more about not actually saying that this "is a mistake". That's already implied by reporting that a lint rule has been violated. As far as I know the other messages don't add this is likely a mistake either.

Reply:
Sounds reasonable.